enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,397 @@
+
+- `approvers`: All enhancements must be approved, but the appropriate people to approve a given
+  enhancement depends on its scope. If an enhancement is limited in scope to a given team or
+  component, then a peer or lead on that team or pillar is an appropriate approver. If an
+  enhancement captures something more broad in scope, then a member of the OpenShift architects team
+  or someone they delegate would be appropriate. Examples would be something that changes the
+  definition of OpenShift in some way, adds a new required dependency, or changes the way customers
+  are supported. Use your best judgement to determine the level of approval needed. If you’re not
+  sure, just leave it blank and ask for input during review.
+
+# Subscription Injection Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement will allow any workload to access RHEL subscription content via a pod annotation.
+
+## Motivation
+
+Containers can download RHEL content if their hosts have subscriptions attached. In 3.x, all nodes
+were assumed to be RHEL 7 nodes that were subscribed individually. The information needed to access
+subscription content (entitlement keys and `redhat.repo` configurations) was symbolically linked
+into the running container at a well-known location that yum/dnf could find. This capability existed
+in our patch of Docker, and was carried over into Red Hat’s container toolchain used on OpenShift 4
+(cri-o, podman, buildah, etc.).
+
+In OpenShift 4 the preferred operating system (RHCOS) is not capable of attaching subscriptions
+individually. To access subscription content, containers must be provided entitlement keys and
+`redhat.repo` configurations via other means. Enabling this capability needs to be simple and
+straightforward.
+
+### Goals
+
+1. Inject entitlement keys and `redhat.repo` configurations into a pod spec.
+2. Define APIs and conventions that allow cluster admins to add the required information to a
+   cluster (entitlement keys and `redhat.repo` configurations).
+
+### Non-Goals
+
+1. Automatically install entitlement keys and `redhat.repo` configurations onto a subscribed
+   cluster.
+
+## Proposal
+
+### User Stories
+
+#### Add debug tools to containers
+
+As a developer debugging my application I want access to RHEL subscription content in my containers
+So that I can install debugging tools from Red Hat via `yum install`
+
+#### Access appropriate RHEL content for container
+
+As a developer using OpenShift I want to be able to access appropriate RHEL content for my container
+So that I download RHEL 8 content in UBI 8 containers And download RHEL 7 content in UBI 7
+containers
```

Comment:
Host subscription plus librhsm in the container solves this.  Based on the entitlement certificate and the RHEL version inside the container, librhsm creates an appropriate `redhat.repo` file for the container version running.

Reply:
@jamescassel is librhsm available in RHCOS, or is this something that exists in UBI containers?

Not having to provide or generate `redhat.repo` out of the box would be very beneficial.

Reply:
It's part of all the UBI and RHEL containers and derivatives, including `ubi-minimal` 7 and 8 containers with `microdnf` instead of `yum`.  RH has done a great job ensuring these containers "just work" on a subscribed host.

Reply:
excellent! I will update the EP to reflect this so that providing `redhat.repo` is optional. My understanding is that Satellite-subscribed clusters will need to provide their own `redhat.repo` configurations.